src/__experimental__/components/radio-button/radio-button.spec.js

```diff
@@ -1,88 +1,81 @@
 import React from 'react';
-import TestRenderer from 'react-test-renderer';
+import { act } from 'react-dom/test-utils';
 import { mount } from 'enzyme';
-import 'jest-styled-components';
-import { css } from 'styled-components';
-import { RadioButton, OriginalRadioButton } from '.';
+import { css, ThemeProvider } from 'styled-components';
+import TestRenderer from 'react-test-renderer';
+import { RadioButton, RadioButtonGroup } from '.';
 import FieldHelpStyle from '../field-help/field-help.style';
 import LabelStyle from '../label/label.style';
-import CheckableInput from '../checkable-input/checkable-input.component';
 import HiddenCheckableInputStyle from '../checkable-input/hidden-checkable-input.style';
 import { StyledCheckableInput } from '../checkable-input/checkable-input.style';
 import StyledCheckableInputSvgWrapper from '../checkable-input/checkable-input-svg-wrapper.style';
 import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
 import guid from '../../../utils/helpers/guid';
 import baseTheme from '../../../style/themes/base';
-import classicTheme from '../../../style/themes/classic';
+import classic from '../../../style/themes/classic';
+import { getValidationType } from '../../../components/validations/with-validation.hoc';
+import small from '../../../style/themes/small';
 
 jest.mock('../../../utils/helpers/guid');
 guid.mockImplementation(() => 'guid-12345');
 
-function render(props, renderer = TestRenderer.create) {
+function render(props = {}, theme = small, renderer = mount) {
+  const {
+    hasError, hasInfo, hasWarning, ...buttonProps
+  } = props;
+  const groupProps = {
+    hasError, hasInfo, hasWarning
+  };
   return renderer(
-    <RadioButton
-      name='my-radio'
-      value='test'
-      { ...props }
-    />
+    <ThemeProvider theme={ theme }>
+      <RadioButtonGroup { ...groupProps }>
+        <RadioButton
+          name='my-radio'
+          value='test'
+          { ...buttonProps }
+        />
+      </RadioButtonGroup>
+    </ThemeProvider>
   );
 }
 
-function renderClassic(props) {
-  return TestRenderer.create(
-    <RadioButton
-      name='my-radio'
-      theme={ classicTheme } value='test'
-      { ...props }
-    />
-  );
-}
+const getRadioButton = wrapper => wrapper.find(RadioButton);
 
-describe('RadioButton', () => {
-  describe('tabindex', () => {
-    describe('when checked === true', () => {
-      it('sets the tabindex attribute to 0', () => {
-        const input = render({ checked: true }, mount).find(CheckableInput);
+const renderClassic = props => render(props, classic);
```

Comment:
do we need this method? We could just use `render` with a second argument.

Reply:
This is purely to make the code easier to read instead of passing around `classic` in the test.